### PR TITLE
Pushes queues

### PR DESCRIPTION
**Pull request by mosheka**
Break pushes using queue
A lot of code cleanup in pushes

### CODE REVIEW

File: protected/classes/push_class.php [hunk 1]
```diff
@@ -5,280 +5,133 @@
...
-class push_class extends staticBase
-{
-    private static $task;
-    public static $coef = 350;
-    public static $count = 0;
-    public static $servers = [];
-    public static $models = [];
+class push_class extends staticBase {
+    public static function sendPushes() {
+        global $cnf;
```
Comment (novichkovv) [inline protected/classes/push_class.php]: should  be replaced

File: protected/classes/push_class.php [hunk 2]
```diff
@@ -5,280 +5,133 @@
...
+        $q = new Q($cnf['q_pushes']);
```
Comment (novichkovv) [inline protected/classes/push_class.php]: $q = new Q('q_pushes');

File: protected/classes/task_class.php [hunk 3]
```diff
@@ -129,15 +129,14 @@ function enrichUsers($users) {
...
 
-		if($pushes) {
-			self::model('pushes')->insertRows($pushes);
-			$q = self::model('task_queue')->getById(self::$task['queue_id']);
-			$got_content = $q['got_content'] + count($pushes);
-			self::model('task_queue')->insert([
-				'id' => $q['id'],
-				'got_content' => $got_content
-			]);
+		$q = new Q($cnf['q_units']);
```
Comment (novichkovv) [inline protected/classes/task_class.php]: $q = new Q('q_units');

File: protected/classes/units_class.php [hunk 4]
```diff
@@ -14,21 +14,33 @@ class units_class extends staticBase
 
     public static function proceedUnits() {
         set_time_limit(0);
+        global $cnf;
+        $q = new Q($cnf['q_units']);
```
Comment (novichkovv) [inline protected/classes/units_class.php]: q_units, remove global

File: protected/classes/units_class.php [hunk 5]
```diff
@@ -14,21 +14,33 @@ class units_class extends staticBase
...
+        }
+        unset($q);
+    }
+
+    public static function processUnit($unit) {
+        $unit['feed_url'] = self::model('feeds')->getById($unit['feed_id'])['url'];
+        $method = 'multi' . ucfirst($unit['feed_key']);
+        $result = self::$method(array($unit));
+        
 		global $cnf;
```
Comment (novichkovv) [inline protected/classes/units_class.php]: remove

File: protected/classes/units_class.php [hunk 6]
```diff
@@ -14,21 +14,33 @@ class units_class extends staticBase
...
-		$q = new Q($cnf['q_units']);
-		while (($units = $q->Get() != array())) {
-			foreach ($units as $id => $unit) {
-				self::processUnit($unit);
-				$q->Delete($id);
-			}
-		}
+		$q = new Q($cnf['q_pushes']);
```
Comment (novichkovv) [inline protected/classes/units_class.php]: $q = new Q('q_pushes');

File: protected/classes/push_class.php (hunk 1 above)
Comment (mosheka) [reply protected/classes/push_class.php]: Done

File: protected/classes/units_class.php (hunk 5 above)
Comment (mosheka) [reply protected/classes/units_class.php]: Done

**Comment by mosheka** [comment]
@novichkovv Please review again

**Pull request merged**